src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportControlCheckTest.java

```diff
@@ -182,4 +182,45 @@ public void testWrongFormatURI() throws Exception {
                 + "InputImportControl.java"), expected);
     }
 
+    @Test
+    public void testUrl() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("url", getURI("imports" + File.separator
+                + "import-control_one.xml"));
+        final String[] expected = {"5:1: " + getCheckMessage(MSG_DISALLOWED, "java.io.File")};
+
+        verify(checkConfig, getPath("imports" + File.separator
+                + "InputImportControl.java"), expected);
+    }
+
+    @Test
+    public void testUrlBlank() throws Exception {
+        final DefaultConfiguration checkConfig = createCheckConfig(ImportControlCheck.class);
+        checkConfig.addAttribute("url", "");
+        final String[] expected = {"1:40: " + getCheckMessage(MSG_MISSING_FILE)};
+
+        verify(checkConfig, getPath("imports" + File.separator
+                + "InputImportControl.java"), expected);
+    }
+
+    @Test(expected = CheckstyleException.class)
```

Comment:
In this particular case the best solution I see is:
```java
        try {
            verify(checkConfig, getPath("imports" + File.separator
                    + "InputImportControl.java"), expected);
        }
        catch (final CheckstyleException ex) {
            return;
        }
        fail();
```
Basically, it only narrows error scope to `verify` method, it doesn't guarantee that we have caught correct exception. 
I have two tests for two different situations. In both cases error is re-thrown with message similar to `cannot initialize module com.puppycrawl.tools.checkstyle.TreeWalker - Cannot set property 'url' to 'https://{WrongCharsInURL}' in module com.puppycrawl.tools.checkstyle.checks.imports.ImportControlCheck`.
This message doesn't contain anything from original error message and doesn't allow us to identify if we caught correct exception.